extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js

```diff
@@ -119,18 +127,32 @@ class AmpFlyingCarpet extends AMP.BaseElement {
     );
   }
 
+  /** @override */
   layoutCallback() {
     try {
-      this.assertPosition();
+      this.assertPosition_();
     } catch (e) {
       // Collapse the element if the effect is broken by the viewport location.
       this./*OK*/collapse();
       throw e;
     }
     this.scheduleLayout(this.children_);
+    listen(this.element, 'amp:built', this.layoutBuiltChild_.bind(this));
```

Comment:
I see, unlike things in sticky-ad, flying-carpet can have multiple different elements as children, which makes it not possible to use `listenOnce`.